### PR TITLE
Acceptance Tests for IAM Policy Resource

### PR DESCRIPTION
**Pull request by Vasundharashukla**
1. **What this PR does / why we need it**:
This PR adds acceptance tests for IAM policy resource.

2. **Additional information**: Tested for all cases.

3. **Screenshot of test results**:
Screenshot 2022-09-09 at 2 13 56 PM
Screenshot 2022-09-09 at 2 14 12 PM

### CODE REVIEW

File: internal/resources/iampolicy/resource_iam_policy_test.go [hunk 1]
```diff
@@ -0,0 +1,598 @@
...
+	WorkspaceResourceVar       string
+	WorkspaceName              string
+	WorkspaceRole1             string
+	NamespaceResourceName      string
+	NamespaceResource          string
+	NamespaceResourceVar       string
+	NamespaceName              string
+	NamespaceRole1             string
+	IAMPolicyResource          string
+	IAMPolicyResourceVar1      string
```
Comment (shreyassreenivas) [inline internal/resources/iampolicy/resource_iam_policy_test.go:96]: can you have them as an array ? 
and so many values, are they needed ? can you group them ?

File: internal/resources/iampolicy/resource_iam_policy_test.go [hunk 2]
```diff
@@ -0,0 +1,598 @@
...
+			{
+				Config: testConfig.getTestIAMPolicyResourceBasicConfigValue(namespaceScope, false),
+				Check:  testConfig.checkIAMPolicyResourceAttributes(namespaceScope),
+			},
+		},
+	},
+	)
+
+	t.Log("IAM policy resource acceptance test complete for single resource!")
+	time.Sleep(2 * time.Minute)
```
Comment (shreyassreenivas) [inline internal/resources/iampolicy/resource_iam_policy_test.go:190]: why sleep??

Comment (Vasundharashukla) [reply internal/resources/iampolicy/resource_iam_policy_test.go:225]: For safety reasons, in case the resource is not deleted from the DB.

Comment (shreyassreenivas) [reply internal/resources/iampolicy/resource_iam_policy_test.go:225]: two mins is a long time for a test to wait.

Comment (shreyassreenivas) [reply internal/resources/iampolicy/resource_iam_policy_test.go:225]: If you want to wait for the resources to be cleaned then validate them through APIs, sleep is not the suggested way

**Comment by shreyassreenivas** [comment]
Can you please post the results of the acceptance test

File: internal/resources/iampolicy/resource_iam_policy_test.go (hunk 2 above)
Comment (shreyassreenivas) [reply internal/resources/iampolicy/resource_iam_policy_test.go:225]: In the interest of time, can you added a TODO which let goes the hardcoded sleep and enhance the acceptance in a different MR. 
please merge the MR once the acceptance results are posted.

**Review by shreyassreenivas — APPROVED** [review]
Merge the changes once the acceptance test results are posted

**Review by ishangupta-ds — APPROVED** [review]
LGTM

**Comment by ishangupta-ds** [comment]
Changes look good, you may merge it.

**Pull request merged**